guardian/gateway
# Forward user IP address in Okta API calls

## What does this change?

Okta by default uses the public IP of the application as the client IP for requests to Okta APIs. However it supports the `X-Forwarded-For` header to forward the originating client's IP address.

This PR sets up all calls to Okta to use the user (client) ip (we get from express `req.ip`) instead of gateway's IP address.

This should help identifying problematic IP addresses in places like the system log and rate limits, and may also allow for better usage of okta's built in threat insight and analysis.

Documentation is available here: https://developer.okta.com/docs/reference/core-okta-api/#ip-address

## How to test

- Deploy to CODE
- Check that when you make requests, that your IP is showing up in the system log/rate limits
